.github/workflows/ci.yml

```diff
@@ -39,13 +39,21 @@ jobs:
         run: |
           .github/workflows/setup-ssh-localhost.sh
           ssh -v localhost
-      - name: Install dependencies (including dev dependencies at frozen version)
+
+      - name: Set up uv
+        uses: astral-sh/setup-uv@v5.2.1
+        with:
+          version: 0.5.22
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies from uv lock
+        if: ${{ inputs.from-lock == 'true' }}
+        # NOTE: We're asserting that the lockfile is up to date
+        run: uv sync --locked --extra dev --extra progressbar --extra optionaltests --extra examples
+        shell: bash
+      - name: Install disk-objectstore 
         # I'm using pip install -e to make sure that the coverage properly traces the runs
         # also of the concurrent tests (maybe we can achieve this differently)
-        run: |
-          python -m pip install --upgrade pip
-          pip install -e .[progressbar,optionaltests]
-          pip install -r requirements.lock
+        run: uv pip install -e .[dev,progressbar,optionaltests,examples]
```

Comment:
Again here `examples` I am imitating the content of the `requirements.lock` that contains packages of all these dependencies. I think it is bad dep management of the extras that the `requirements.lock` contains everything and was used here for just running the tests but I am just imitating. We can improve this an another PR if you want.